specs/vep-554-apache-airflow-integration/README.MD

```diff
@@ -0,0 +1,401 @@
+* **DAG**: Directed Acyclic Graph
+
+## Motivation
+
+Тhe need for multi-job non-linear analytics is growing. Users need to be able to define
+dependencies between the units of work (tasks or jobs). For instance, they need to ingest
+from multiple systems into Data Lake to model the data. All lake tables are used
+afterwards by subsequent processing jobs to generate Data Marts. The processing jobs
+start only after the ingestion jobs are over.
+
+Currently, VDK supports only multi-step linear analytics by defining sequential steps
+within a single process (job). The steps can be written either in SQL or Python. Due to
+the fact that the steps are linear and with limited ability to run in parallel, more
+complex non-linear analytics cannot be done.
+
+There are two workarounds to solve this:
+
+1. Ingestion and processing jobs can be defined as sequential steps in a single job.
+   However, this approach has several drawbacks:
+    * The steps are sequential with limited ability to run in parallel.
+    * The execution time of the Data Job is limited (currently up to 12 hours).
+
+2. Ingestion and processing jobs can be scheduled at a certain time with an appropriate
+   time offset between them (e.g. the ingestion jobs start at 2 p.m., as the processing
+   jobs starts 8 p.m.). However, this approach is not reliable as the users need to know
+   how long the jobs will take, some jobs may take more time than usual.
+
+The support for the definition of dependencies between the jobs will be introduced, in
+order to provide the ability to do multi-job non-linear analytics to our users. It will
+be achieved through the integration of VDK with Apache Airflow.
+
+Airflow has been chosen because it is a tool that can create, organize, and monitor
+workflows. It is open-source, hence it is free and has a wide range of support as well.
+It is one of the most trusted platforms that is used for orchestrating workflows and is
+widely used and recommended by top data engineers. This tool provides many features like
+a proper visualization of data pipelines and workflows, workflow status, data logs, and
+codes in detail. Airflow is a distributed system that is highly scalable and can be
+connected to various sources, which makes it flexible. These features allow it to be used
+efficiently in the orchestration of complex workflow and data pipelining problems.
+
+## Requirements and Goals
+
+### Goals
+
+#### Provide an ability to define dependencies between units of work (tasks or jobs)
+
+A user wants to start a second job after the first job is over. For example, the user has
+several jobs, after each job is complete, the user would like to start a common
+post-processing job.
+
+Use cases:
+
+* Sync between multiple synchronous ingestion and processing jobs - the user needs to be
+  able to ingest from multiple systems into Data Lake to model the data. All lake tables
+  are used afterwards by subsequent processing jobs to generate Data Marts. The
```

Comment:
Define Data Mart. It's a bit of ambiguous term in the data community. And here we mean a database with a set of tables simply.